src/main/scala/Main.scala

```diff
@@ -71,20 +64,90 @@ object Main {
         esIndexer = new ESIndexer(argMap("esurl"))
         s3Downloader = new S3Downloader(argMap("bucket"))
 
-        if(argMap("do").equals("adminremote")) {
-            adminIndexFilesRemote(argMap("starturl"), argMap("midurl"), argMap("endurl"))
-        } else if(argMap("do").equals("adminlocal")) {
-            adminIndexFilesLocal(argMap("localinput"))
+        /**
+          * Transforms a call-by-name into a no-parameter function call to be able to use it in a varargs
+          *
+          * https://stackoverflow.com/questions/13307418/scala-variable-argument-list-with-call-by-name-possible
+          *
+          * @param a the call-by-name parameter
+          * @tparam A its return type
+          * @return the call by name parameter as a function
+          */
+        implicit def bynameToNoarg[A](a: => A): () => A = () => a
+        /**
+          * Syntactic sugar to parallelize the execution of multiple function calls.
+          *
+          * Creates a Future per block of code to execute and then blocks until all the tasks are done.
+          *
+          * https://alvinalexander.com/scala/fp-book/how-to-use-by-name-parameters-scala-functions
+          *
+          * @param codes the function calls to parallelize
+          */
+        case class Tasks(codes: (() => Any)*) {
+            Await.result(
+                Future.traverse(codes) { code =>
+                    Future[Unit] {
+                        println(code())
+                    }
+                }
+                , Duration.Inf)
+        }
 
-        } else printHelp
+        val startTime = System.currentTimeMillis()
+
+        Tasks(
+            indexParticipants("https://kf-api-dataservice.kids-first.io", "/participants", "limit=100&visible=true"),
+            if(argMap("do").equals("adminremote")) {
+                indexPDFRemote(argMap("starturl"), argMap("midurl"), argMap("endurl"))
+            } else if(argMap("do").equals("adminlocal")) {
+                indexPDFLocal(argMap("localinput"))
+            }
+        )
+
+        esIndexer.cleanup()
+
+        println("took "+(System.currentTimeMillis()-startTime)/1000+" seconds")
+        System.exit(0)
     }
 
-    def adminIndex(pdf: InputStream, name: String, typ: String = "local"): String = {    //do: OCR -> NLP -> ES
+    def indexParticipants(start: String, mid: String, end: String): Unit = {
+        Await.result(
```

Comment:
I don't like that the three Index methods in Main have the same ``Await.result(
Future.sequence of Future.traverse``. Do you have an idea on how to change that?

Reply:
Fait un fix avec ma 2e PR: à voir